spec-0008/index.md

```diff
@@ -0,0 +1,227 @@
+
+With the creation of the [Supply-chain Levels for Software Artifacts (SLSA) framework](https://slsa.dev/) and [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) standard being more widely adopted there are now high level developer tools, maintained by professional security teams, that are available with clear recommendations on how to use them.
+
+This SPEC outlines the requirements for practically adopting these security tools and recommendations with a focus on securely publishing release artifacts.
+Securely building release artifacts will be covered in a later SPEC. This set of recommendations complements the recommendations from [SPEC 6 — Keys to the Castle](https://github.com/scientific-python/specs/blob/main/spec-0006/index.md).
+
+This SPEC is written with GitHub in mind, note these recommendations still apply with other services such as GitLab and we refer project to these services' documentations and guidelines.
+
+## Implementation
+
+With a focus on securing the release artifact distribution process, the following processes and standards should be adopted.
+
+### Document the release process
+
+The release process should be clearly and fully documented in the developer documentation and describe each step to make a release and the permissions required to do so.
+
+### Hardening workflow environment permissions
+
+Workflows that publish release artifacts should have _run triggers_ that require intentional actions by maintainers (e.g., `on workflow_dispatch` in GitHub Actions) and require multiple maintainers to approve the workflow to run (c.f. "Use GitHub Actions environments" section below).
+This is to safeguard the project from any one maintainer having the ability to commit to the default branch and make a release directly.
+
+It is also strongly recommended that the repository requires signed commits so that all releases have a verified commit to which they correspond.
+The branch from which the release is made should also be protected.
+
+#### Restrict permissions in CI runners to the minimum required
+
+To restrict the attack surface area of arbitrary code execution in CI runners, the default permissions the runners have should be restricted to the minimum possible (read access). In the GitHub Action workflow, this is accomplished by defining the following workflow global permissions block before any jobs are defined.
+
+```yaml
+permissions:
+  content: read
+```
+
+Elevating permissions beyond this should be done at the job level by redefining the permissions block in the job.
+
+#### Restrict permitted actions in workflows
+
+GitHub allows to restrict the actions that workflows can use via repositories settings at `https://github.com/ORG/PROJECT/settings/actions`.
```

Comment:
I can't access the link so not sure, but at least for scikit-image it's definitely `ORG/REPO`. Is it possible that the link above refers to the organization settings for the `scikit-learn` org? It seems like you can at least set some of the settings at org-level as well. So we should probably address this to some extent.

Reply:
Yeah, just checked, your link above doesn't refer to the PROJECT but the ORG.

Reply:
Indeed. I thought that the goal was to use the organisation level settings page. So maybe the thing to do is to make it more obvious to the reader which of the two it is :D

Reply:
> So maybe the thing to do is to make it more obvious to the reader which of the two it is :D

@betatim We tried to do this with making `ORG` and `PROJECT` instead of `<ORG>` and `<PROJECT>` and that we lead with

> workflows can use via repositories settings

or perhaps it would be even more clear with `$ORG` and `$PROJECT`.

So I'll do that and then if this is visually ugly to people we can revert this.